eultheme/middleware.py

```diff
@@ -0,0 +1,23 @@
+from django.conf import settings
+from django.shortcuts import render
+from downtime.middleware import DowntimeMiddleware
+
+class DownpageMiddleware(DowntimeMiddleware):
+
+    def process_request(self, request):
+        def get_client_ip(request):
```

Comment:
I think it would make more sense (and be more readable) to define this as a class method rather than redefining every time we process a request.  I _think_ it would be more efficient too, but I don't actually know.